### PR TITLE
Add get source feature

### PR DESCRIPTION
**Pull request by steve0hh**
PR for issue #79, support for getting the current page source

and

PR for issue #74, setting the recv_timeout to infinity

### CODE REVIEW

**Comment by steve0hh** [comment]
f209199 also fixes a big where the config variable js_errors is cannot be set to false.

**Comment by keathley** [comment]
Thanks. I'd like to separate these issues into their own branches and merges. I'll just cherry pick your commits for the httpoison timeout and the js_errors issue. That way we can focus on just the logic for getting page source in this PR :+1:

File: lib/wallaby.ex [hunk 1]
```diff
@@ -46,9 +46,13 @@ defmodule Wallaby do
   end
 
   def js_errors? do
-    Application.get_env(:wallaby, :js_errors) || true
+    Application.get_env(:wallaby, :js_errors)
```
Comment (keathley) [inline lib/wallaby.ex]: `Application.get_env/3` actually accepts a default (which is what I should have used in the first place). Thanks for catching my silly mistake.

Comment (steve0hh) [inline lib/wallaby.ex]: Oh cool!

Didn't know that. Another new thing learnt today. 😃

**Comment by keathley** [comment]
The other fixes have been merged into master. If you could rebase your branch so that its up to date with master that would be rad.

Otherwise, I think the only other thing that this PR needs is a test. The way that I would do it is to create a new session test like: `test/wallaby/session/page_source_test.exs`. Then I would write a very basic test like:

```elixir
test "page_source/1 retrieves the source of the current page" do
  source = 
    session
    |> visit(server.base_url <> "/index.html")
    |> page_source
  
  html_path = Path.absname("test/support/pages/index.html", Application.app_dir(:wallaby))
  assert source == File.read!(html_path)
end
```

You'll probably have to adjust that test to work in the real world. But it should be close. You'll also need to make sure that the test case is `async: false` since its interacting with the outside world. Let me know if you have any issues :+1:

File: lib/wallaby/driver.ex [hunk 2]
```diff
@@ -256,6 +256,16 @@ defmodule Wallaby.Driver do
     resp["value"]
   end
 
+  @doc """
+  Retrives the current page source from session
+  """
+  def source(session) do
```
Comment (keathley) [inline lib/wallaby/driver.ex]: Lets change this to `page_source` since it can only ever be used to get the source of the page.

**Comment by casellepete** [comment]
Thank you for your efforts on this.  
Looking forward to it.

**Comment by keathley** [comment]
@steve0hh are you still interested in finishing this up? If not thats cool. I can take it from here. But I thought I would check first.

**Comment by steve0hh** [comment]
Hey @keathley , sorry about the delay. 

Was busy with school and needed to figure out how to merge upstream with my own fork so there's no conflicts. 😄

**Comment by opsb** [comment]
Would be great to get `page_source` merged in, seems like a simple PR and would be happy to do it myself. @steve0hh perhaps you could put a PR together for just https://github.com/keathley/wallaby/pull/81/files#diff-58dd4483c60c6311fba480cefc54fbc6R266 ?

**Comment by keathley** [comment]
No worries @steve0hh. Typically the best way to update your branch is to do a [rebase](https://git-scm.com/docs/git-rebase) instead of a merge. You can also use rebase to "squash" commits together or re-order commits. It helps keep the commit log clean and linear. Thanks for the work on this :+1:

@opsb I'll just cherry-pick the commit and do some cleanup.

**Comment by keathley** [comment]
This has been merged here: 6f0e6babbd7862f6546b7601f74225ca3ccc7d93

**Comment by keathley** [comment]
Thanks @steve0hh :tada:

**Pull request closed without merging**